src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -175,9 +179,11 @@ internal static bool IsInConstructorParameterScope(int position, ConstructorDecl
                     IsBetweenTokens(position, initializerOpt.ColonToken, nextToken);
             }
 
+            Debug.Assert(constructorDecl.Body != null || (constructorDecl.ExpressionBody != null && constructorDecl.SemicolonToken!= null), 
```

Comment:
I wanted to make sure that the body is fine, before passing it to `IsBetweenTokens`. But that seems to be wrong in case the semicolon is missing? Will review it carefully.